pkg/rosa/oauth.go

```diff
@@ -0,0 +1,109 @@
+package rosa
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	restclient "k8s.io/client-go/rest"
+)
+
+type TokenResponse struct {
+	AccessToken string
+	ExpiresIn   time.Duration
+}
+
+func RequestToken(apiURL, username, password string, config *restclient.Config) (*TokenResponse, error) {
+	clientID := "openshift-challenging-client"
+	oauthUrl, err := buildOauthUrl(apiURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build oauth url: %w", err)
+	}
+
+	tokenReqUrl := fmt.Sprintf("%s/oauth/authorize?response_type=token&client_id=%s", oauthUrl, clientID)
+	request, err := http.NewRequest(http.MethodGet, tokenReqUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("Authorization", getBasicHeader(username, password))
+	request.Header.Set("X-CSRF-Token", "1")
```

Comment:
What's this?

Reply:
this is a required header by the openshift oauth server to prevent CORS errors.
see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/authentication_and_authorization/understanding-authentication#oauth-token-requests_understanding-authentication

Reply:
Could we add a comment with link?

Reply:
sure